src/pages/Swap/index.js

```diff
@@ -61,6 +68,11 @@ class Swap extends Component {
       });
   }
 
+  shouldComponentUpdate(nextProps, nextState) {
+    return !deepEqual(nextProps, this.props) ||
+      !deepEqual(nextState, this.state);
+  }
+
```

Comment:
@chikeichan Added this shouldcomponentupdate function.

When I was testing the approval process, the app would always freeze after I confirmed with Metamask (and therefore the app never enters the 'Pending' state). After a long time debugging, it turns out the react components were rendering basically nonstop during the pending state, probably due to drizzle. Looked into it and the drizzle docs suggest adding a `shouldComponentUpdate`. added it and it works perfectly now. This probably fixes #49 also

Reply:
![](https://media.giphy.com/media/CwAlFqwEGreoM/giphy.gif)